apps/native/app/src/screens/wallet-pass/wallet-pass.tsx

```diff
@@ -214,21 +264,64 @@ export const WalletPassScreen: NavigationFunctionComponent<{
     }
   }
 
-  const fields = data?.payload?.data ?? []
-  const hasPkpass = data?.license?.pkpass
-  const hasValidPkpass =
-    data?.license?.pkpassStatus === GenericUserLicensePkPassStatus.Available
-  // this is coming soon.. disable add button if not true.
-  // const hasValidatedPkpass = data?.license?.pkpassValidation;
+  const expirationTimeCallback = useCallback(() => {
+    void res.refetch()
+  }, [])
+
+  const expirationTime = useMemo(() => {
+    const exp = data?.barcode?.exp
+
+    if (exp) {
+      const expirationTime = new Date(exp)
+      // We subtract 5 seconds to make sure the barcode is still valid when switching to a new barcode
+      expirationTime.setSeconds(expirationTime.getSeconds() - 5)
+
+      return expirationTime
+    }
+  }, [data?.barcode?.exp])
 
   return (
     <View style={{ flex: 1 }}>
       <View style={{ height: cardHeight }} />
-      <Information contentInset={{ bottom: 162 }}>
-        <SafeAreaView style={{ marginHorizontal: 16 }}>
+      <LicenseCardWrapper>
+        <LicenseCard
+          nativeID={`license-${licenseType}_destination`}
+          type={licenseType}
+          logo={
+            data?.license?.type === GenericLicenseType.DriversLicense
+              ? getImageFromRawData(data?.payload?.rawData)
+              : undefined
+          }
+          date={new Date(Number(data?.fetch?.updated))}
```

Comment:
new Date(Number(undefined)) = Invalid Date

Reply:
Ahh this is not from this PR, but I will fix :)